File: frontend/lib/presets.ts
import type { PlatformPreset } from '@/types'

export interface PresetInfo {
  width: number
  height: number
  label: string
}

export const PLATFORM_PRESETS: Record<PlatformPreset, PresetInfo> = {
  instagram_post:       { width: 1080, height: 1080, label: 'Instagram Post' },
  instagram_story:      { width: 1080, height: 1920, label: 'Instagram Story' },
  instagram_reel_cover: { width: 1080, height: 1920, label: 'Instagram Reel Cover' },
  facebook_post:        { width: 1200, height:  630, label: 'Facebook Post' },
  facebook_cover:       { width:  820, height:  312, label: 'Facebook Cover' },
  facebook_story:       { width: 1080, height: 1920, label: 'Facebook Story' },
  twitter_post:         { width: 1200, height:  675, label: 'Twitter Post' },
  twitter_header:       { width: 1500, height:  500, label: 'Twitter Header' },
  linkedin_post:        { width: 1200, height:  627, label: 'LinkedIn Post' },
  linkedin_banner:      { width: 1584, height:  396, label: 'LinkedIn Banner' },
  tiktok_video_cover:   { width: 1080, height: 1920, label: 'TikTok Video Cover' },
  youtube_thumbnail:    { width: 1280, height:  720, label: 'YouTube Thumbnail' },
  youtube_banner:       { width: 2560, height: 1440, label: 'YouTube Banner' },
}

export const PRESETS_BY_PLATFORM: Record<string, PlatformPreset[]> = {
  Instagram: ['instagram_post', 'instagram_story', 'instagram_reel_cover'],
  Facebook:  ['facebook_post', 'facebook_cover', 'facebook_story'],
  'Twitter/X': ['twitter_post', 'twitter_header'],
  LinkedIn:  ['linkedin_post', 'linkedin_banner'],
  TikTok:    ['tiktok_video_cover'],
  YouTube:   ['youtube_thumbnail', 'youtube_banner'],
}

File: backend/app/services/presets.py
import re
from datetime import datetime, timezone

PLATFORM_PRESETS: dict[str, tuple[int, int, str]] = {
    "instagram_post":       (1080, 1080, "Instagram Post"),
    "instagram_story":      (1080, 1920, "Instagram Story"),
    "instagram_reel_cover": (1080, 1920, "Instagram Reel Cover"),
    "facebook_post":        (1200,  630, "Facebook Post"),
    "facebook_cover":       ( 820,  312, "Facebook Cover"),
    "facebook_story":       (1080, 1920, "Facebook Story"),
    "twitter_post":         (1200,  675, "Twitter Post"),
    "twitter_header":       (1500,  500, "Twitter Header"),
    "linkedin_post":        (1200,  627, "LinkedIn Post"),
    "linkedin_banner":      (1584,  396, "LinkedIn Banner"),
    "tiktok_video_cover":   (1080, 1920, "TikTok Video Cover"),
    "youtube_thumbnail":    (1280,  720, "YouTube Thumbnail"),
    "youtube_banner":       (2560, 1440, "YouTube Banner"),
}

PRESET_TO_ASPECT_RATIO: dict[str, str] = {
    "instagram_post":       "1:1",
    "instagram_story":      "9:16",
    "instagram_reel_cover": "9:16",
    "facebook_story":       "9:16",
    "tiktok_video_cover":   "9:16",
    "facebook_post":        "16:9",
    "twitter_post":         "16:9",
    "linkedin_post":        "16:9",
    "youtube_thumbnail":    "16:9",
    "twitter_header":       "16:9",
    "facebook_cover":       "16:9",
    "linkedin_banner":      "16:9",
    "youtube_banner":       "16:9",
}

MODEL_FOR_PROVIDER: dict[str, str] = {
    "openai": "gpt-image-2",
    "gemini": "gemini-3-pro-image-preview",
}

_BRAND_NAME_NON_ALNUM = re.compile(r"[^a-z0-9]+")


def sanitize_brand_name(raw: str) -> str:
    cleaned = _BRAND_NAME_NON_ALNUM.sub("-", raw.lower()).strip("-")
    if not cleaned:
        return "brand"
    truncated = cleaned[:40].rstrip("-")
    return truncated or "brand"


def build_download_filename(
    brand_name: str, preset_identifier: str, completed_at: datetime
) -> str:
    ts = completed_at.astimezone(timezone.utc).strftime("%Y%m%d-%H%M%S")
    return f"{sanitize_brand_name(brand_name)}-{preset_identifier}-{ts}.png"
